File: servers/local-memory-indexer/tests/fixtures/repos/ts-sample/src/math.ts
export function add(a: number, b: number): number {
  const result = a + b;
  return result;
}

export function subtract(a: number, b: number): number {
  const result = a - b;
  return result;
}

export function multiply(a: number, b: number): number {
  return a * b;
}

export class Calculator {
  private _value = 0;

  add(n: number): this {
    this._value += n;
    return this;
  }

  subtract(n: number): this {
    this._value -= n;
    return this;
  }

  multiply(n: number): this {
    this._value *= n;
    return this;
  }

  reset(): this {
    this._value = 0;
    return this;
  }

  result(): number {
    return this._value;
  }
}

File: servers/local-memory-indexer/tests/fixtures/repos/py-sample/math.py
def add(a, b):
    """Add two numbers."""
    result = a + b
    return result


def subtract(a, b):
    """Subtract b from a."""
    result = a - b
    return result


def multiply(a, b):
    """Multiply two numbers."""
    return a * b


class Calculator:
    """A simple calculator with chainable methods."""

    def __init__(self):
        self._value = 0

    def add(self, n):
        self._value += n
        return self

    def subtract(self, n):
        self._value -= n
        return self

    def multiply(self, n):
        self._value *= n
        return self

    def reset(self):
        self._value = 0
        return self

    def result(self):
        return self._value
